### PR TITLE
entity match on all people (not just ones w open bookings) if external_ids are present

### LINKED ISSUE #534: If an external id exists, we should query all bookings
If an external id is provided in a scrape, we should query closed bookings as well as open bookings so we can match people that recidivate within a county.

### PR DESCRIPTION
**Pull request by terinpw**
This will allow us to support scraping historical bookings as long as there is an external_id for the person.

Fixes #534

### CODE REVIEW

File: recidiviz/persistence/database/database.py [hunk 1]
```diff
@@ -55,10 +59,32 @@ def read_bookings(session):
...
+    Reads all people for the given |region| that have external_ids that match
+    the external_ids from the |ingested_people|
+
+    Args:
+        session: The transaction to read from
+        region: The region to match against
+        ingested_people: The ingested people to match against
+    Returns: List of people that match the provided |ingested_people|
+    """
+    external_ids = {p.external_id for p in ingested_people}
```
Comment (arian487) [inline recidiviz/persistence/database/database.py]: Is this meant to be a map?

Comment (jamwalla) [reply recidiviz/persistence/database/database.py]: it's a set, right?

File: recidiviz/tests/persistence/entity_matching_test.py [hunk 2]
```diff
@@ -146,3 +146,43 @@ def test_match_entites(self):
...
+        ingested_closed_booking.booking_id = None
+        ingested_closed_booking.facility = _FACILITY
+
+        ingested_person = database_utils.convert_person(schema_person)
+        ingested_person.person_id = None
+        ingested_person.bookings = [ingested_closed_booking]
+
+        entity_matching.match_entities(Session(), _REGION, [ingested_person])
+
+        # pylint: disable=protected-access
```
Comment (jamwalla) [inline recidiviz/tests/persistence/entity_matching_test.py]: why is this necessary?

File: recidiviz/tests/persistence/entity_matching_test.py [hunk 3]
```diff
@@ -146,3 +146,43 @@ def test_match_entites(self):
...
+                                      birthdate=_DATE,
+                                      place_of_residence=_PLACE_1,
+                                      region=_REGION,
+                                      bookings=[closed_schema_booking])
+
+        session = Session()
+        session.add(schema_person)
+        session.commit()
+
+        # pylint: disable=protected-access
```
Comment (jamwalla) [inline recidiviz/tests/persistence/entity_matching_test.py]: is this leftover code?

File: recidiviz/persistence/database/database.py (hunk 1 above)
Comment (terinpw) [reply recidiviz/persistence/database/database.py]: yup it's a set

File: recidiviz/tests/persistence/entity_matching_test.py (hunk 2 above)
Comment (terinpw) [reply recidiviz/tests/persistence/entity_matching_test.py]: Resolved (and below)
